libpoolprotocols/PoolManager.cpp

```diff
@@ -9,247 +10,223 @@ PoolManager::PoolManager(boost::asio::io_service& io_service, PoolClient* client
+
+    p_client->onConnected([&]() {
+        m_activeConnectionHost = m_connections.at(m_activeConnectionIdx).Host();
         cnote << "Established connection with "
               << (m_connections.at(m_activeConnectionIdx).Host() + ":" +
                      toString(m_connections.at(m_activeConnectionIdx).Port()))
               << " at "
               << p_client->ActiveEndPoint();
 
-		// Rough implementation to return to primary pool
-		// after specified amount of time
-		if (m_activeConnectionIdx != 0 && m_failoverTimeout > 0)
-		{
-
-			m_failovertimer.expires_from_now(boost::posix_time::minutes(m_failoverTimeout));
-			m_failovertimer.async_wait(m_io_strand.wrap(boost::bind(&PoolManager::check_failover_timeout, this, boost::asio::placeholders::error)));
-
-		}
-		else
-		{
-			m_failovertimer.cancel();
-		}
-
-		if (!m_farm.isMining())
-		{
-			cnote << "Spinning up miners...";
-			if (m_minerType == MinerType::CL)
-				m_farm.start("opencl", false);
-			else if (m_minerType == MinerType::CUDA)
-				m_farm.start("cuda", false);
-			else if (m_minerType == MinerType::Mixed) {
-				m_farm.start("cuda", false);
-				m_farm.start("opencl", true);
-			}
-		}
-
-	});
-
-	p_client->onDisconnected([&]()
-	{
-		dev::setThreadName("main");
+        // Rough implementation to return to primary pool
+        // after specified amount of time
+        if (m_activeConnectionIdx != 0 && m_failoverTimeout > 0)
+        {
+            m_failovertimer.expires_from_now(boost::posix_time::minutes(m_failoverTimeout));
+            m_failovertimer.async_wait(m_io_strand.wrap(boost::bind(
+                &PoolManager::check_failover_timeout, this, boost::asio::placeholders::error)));
+        }
+        else
+        {
+            m_failovertimer.cancel();
+        }
+
+        if (!m_farm.isMining())
+        {
+            cnote << "Spinning up miners...";
+            if (m_minerType == MinerType::CL)
+                m_farm.start("opencl", false);
+            else if (m_minerType == MinerType::CUDA)
+                m_farm.start("cuda", false);
+            else if (m_minerType == MinerType::Mixed)
+            {
+                m_farm.start("cuda", false);
+                m_farm.start("opencl", true);
+            }
+        }
+    });
+
+    p_client->onDisconnected([&]() {
+        dev::setThreadName("main");
         cnote << "Disconnected from " + m_activeConnectionHost << p_client->ActiveEndPoint();
 
-		// Clear queue of submission times as we won't get any further response for them (if any left)
-		// We need to consume all elements as no clear mehod is provided.
-		std::chrono::steady_clock::time_point m_submit_time;
-		while (m_submit_times.pop(m_submit_time)) {}
-
-		// Do not stop mining here
-		// Workloop will determine if we're trying a fast reconnect to same pool
-		// or if we're switching to failover(s)
-
-	});
-
-	p_client->onWorkReceived([&](WorkPackage const& wp, bool check_for_duplicates)
```

Comment:
We always check for duplicates. This function is never called unless we've received a new job from the protocol.